src/Adapter/Cart/CartRuleSorter.php

```diff
@@ -0,0 +1,103 @@
+<?php
```

Comment:
all the logic in this file seems to be a duplicate of `CartRule.php`. Is that possible to refactor it?